src/systemTest/java/com/flightstats/hub/model/Links.java

```diff
@@ -0,0 +1,12 @@
+package com.flightstats.hub.model;
+
+import lombok.Builder;
+import lombok.Value;
+
+@Builder
+@Value
+public class Links {
+    private Link self;
+    private Link channel;
+    private String [] uris;
```

Comment:
It feels kind of icky to me to reuse code from the hub in tests that may be testing the validity of that code. I can certainly get over that feeling, is there a convention and a clear line on this?

Reply:
Model classes should be reused in tests. It would be redundant to redefine them in the test and error prone. Data model of the application does not change,.

Reply:
Well, it's more code to maintain, more things to get out of sync if you ever change one of those classes. There's some value in testing with a completely different code base, but that's typically restricted to mission critical things like NASA rover code or some such because it comes at a high cost.

I mean, to go wild on that front, if you really want to avoid using the same code then you'd want to use completely different 3rd party libraries as well. Though, that's not feasible if you're using the same Java core libraries, so then you'd be looking at writing the tests in a completely independent language.
 
Less code is generally better. Less to maintain, less to fix, less bugs created. I'd trust reusing my core domain POJOs like this the same way I'd trust using URI or String.

Reply:
I think one of the problems with many of the hub domain objects is that they're not simple POJOs and have strange responsibilities that aren't just modeling the domain. At that point, it feels a little gross letting that production logic leak into system tests, which should be black box. Another problem is that the hub does some super strange things with manual serialization, so some things that you get in a response don't accurate map to a model in the hub. I'd really love to work towards extracting the domain objects into a library of small lombok objects that specify inputs and outputs of the hub and can be used in both the hub and in clients/consumers.

Reply:
I've gone through and replaced what I could with the hub core model, Links is not available in that model